tests/scale_test/analysis/failure_analysis.py

```diff
@@ -0,0 +1,284 @@
+        'status': status,
+        'task_name': metadata.get('name', '') if status == STATUSES['FAILED'] or status == STATUSES['ABORTED'] else '',
+        'error': get_failure_message(metadata, record_std_err) if status == STATUSES['FAILED'] else ''
+    }
+    return data
+
+
+def main(cromwell_url, auth, headers, output_file, record_std_err=True, start=None, end=None, name=None, statuses=None, labels=None, page_size=None, page=None, expand_subworkflows=True):
+    # Get target workflows
+    result_list_metadata = query_workflows(cromwell_url, auth, headers, start, end, name, statuses, labels, page_size, page)
+
+    # Get workflow metadata
+    result_ids = list(map(lambda dic: dic['id'], result_list_metadata))
+    logging.info('Retrieving Adapter Workflows\' Metadata: ')
+    adapter_metadata = []
+    for idx, workflow_id in enumerate(result_ids):
+        logging.info('Current {0}/{1}: {2}'.format(idx + 1, len(result_ids), workflow_id))
+        workflow_metadata = get_metadata(cromwell_url, workflow_id, auth, headers, expand_subworkflows=expand_subworkflows)
+        adapter_metadata.append(workflow_metadata)
+    adapter_metrics = [parse_workflow(workflow_metadata, cromwell_url) for workflow_metadata in adapter_metadata]
+
+    # Check for duplicate bundles
+    total_bundle_ids = [wf['bundle_id'] for wf in adapter_metrics]
+    find_duplicate_bundle_ids(total_bundle_ids)
+
+    # Group workflows by status
+    summary = sort_workflows_by_status(adapter_metrics)
+
+    workflow_data = []
+    for status in summary.keys():
+        workflow_count = len(summary[status])
+        message = '{}: {}'.format(status, str(workflow_count))
+        if status == STATUSES['SUCCEEDED']:
+            bundle_ids = [wf['bundle_id'] for wf in summary[status]]
+            message += ' ({} unique bundles)'.format(str(len(set(bundle_ids))))
+        logging.info(message)
+        if status != STATUSES['FAILED'] and status != STATUSES['ABORTED']:
+            workflow_data.extend([format_metadata_output(w, record_std_err) for w in summary[status]])
+
+    # Find failed tasks
+    failed_workflows = summary[STATUSES['FAILED']]
+    failures = group_workflows_by_failed_task(failed_workflows, record_std_err=record_std_err)
+    workflow_data.extend(failures)
+
+    # Find last task of aborted workflows
+    aborted_workflows = summary[STATUSES['ABORTED']]
+    aborted = group_workflows_by_failed_task(aborted_workflows, record_std_err=record_std_err)
+    workflow_data.extend(aborted)
+
+    with open(output_file, 'w') as f:
+        f.write('Primary Bundle ID,Workflow ID,Workflow Status,Failed Task,Workflow Error\n')
+        for each in workflow_data:
+            f.write('{},{},{},{},{}\n'.format(each['bundle_id'], each['id'], each['status'],
+                                              each['task_name'], each.get('error', '')))
+
+
+if __name__ == '__main__':
+    logging.basicConfig(level=logging.INFO)
```

Comment:
For this use case (a single script), click could be a good util over than argparse :)

Reply:
@rexwangcc can you clarify the advantages of using click over argparse for a single script? It seems like the script would be invoked the same way.